client/components/LearnerViewDirective.js

```diff
@@ -37,99 +37,103 @@ tie.directive('learnerView', [function() {
                 </div>
               </div>
             </div>
-            <div class="tie-coding-ui">
-              <div class="tie-feedback-window" ng-class="{'night-mode': isInDarkMode}">
-                <div class="tie-feedback">
-                  <p ng-repeat="paragraph in feedbackParagraphs track by $index"
-                      class="tie-feedback-paragraph"
-                      ng-class="{'tie-feedback-paragraph-code': paragraph.isCodeParagraph()}">
-                    <span ng-if="$first">{{feedbackTimestamp}}</span>
-                    <span ng-if="paragraph.isTextParagraph()">
+            <div class="tie-question-ui">
+              <div class="tie-question-window" ng-class="{'night-mode': isInDarkMode}">
+                <div class="tie-greetings">
+                  <p ng-repeat="paragraph in greetingParagraphs track by $index">
                       {{paragraph.getContent()}}
-                    </span>
-                    <span ng-if="paragraph.isCodeParagraph()">
-                      <code-snippet content="paragraph.getContent()"></code-snippet>
-                    </span>
                   </p>
                 </div>
-                <div class="tie-feedback-syntax-error">
+                <h3 class="tie-question-title">{{title}}</h3>
+                <div class="tie-previous-instructions">
+                  <div ng-repeat="previousInstruction in previousInstructions track by $index">
+                    <div ng-repeat="instruction in previousInstruction track by $index">
+                      <p ng-if="instruction.type == 'text'">{{instruction.content}}</p>
+                      <pre class="tie-question-code" ng-class="{'night-mode': isInDarkMode}" ng-if="instruction.type == 'code'">{{instruction.content}}</pre>
+                    </div>
+                    <hr>
+                  </div>
+                </div>
+                <div class="tie-instructions">
+                  <div ng-repeat="instruction in instructions">
+                    <p ng-if="instruction.type == 'text'">{{instruction.content}}</p>
+                    <pre class="tie-question-code" ng-class="{'night-mode': isInDarkMode}" ng-if="instruction.type == 'code'">{{instruction.content}}</pre>
+                  </div>
+                </div>
+                <div class="tie-feedback">
+                  <div ng-repeat="set in feedbackStorage">
+                    <hr>
+                    <span class="tie-feedback-timestamp">{{set.timestamp}}<span>
```

Comment:
I'm thinking we no longer need the timestamp for this UI. The timestamp was introduced in the other UI because feedback was overwritten rather than appended; so, if the same error or feedback was given multiple times consecutively, it would not be apparent to the user. Here, since we are always appending feedback, I don't think we need a timestamp anymore.